File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/resolution/SemanticLexiconEntry.java
package com.foundgine.core.semantic.resolution;

import com.foundgine.core.abstractions.*;
import java.util.List;

public record SemanticLexiconEntry(String canonicalName, SemanticLexicalCandidateKind kind, String searchText,
		EntityId entityId, RelationshipId relationshipId, FieldId fieldId, EntityId sourceEntityId,
		EntityId targetEntityId, String value, List<String> aliases, String description) {
	public SemanticLexiconEntry {
		aliases = aliases == null ? List.of() : List.copyOf(aliases);
	}

	public SemanticLexiconEntry(String canonicalName, SemanticLexicalCandidateKind kind, String searchText) {
		this(canonicalName, kind, searchText, null, null, null, null, null, null, List.of(), null);
	}

	public List<String> effectiveAliases() {
		return aliases;
	}
}

File: src/csharp/Foundgine.Core/Semantic/Resolution/Lexical/SemanticLexiconEntry.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Resolution;

/// <summary>
/// Provider-neutral document that can be projected into a lexical search
/// index. The semantic contract is the source of truth; the index is a derived
/// retrieval projection.
/// </summary>
public sealed record SemanticLexiconEntry(
    string CanonicalName,
    SemanticLexicalCandidateKind Kind,
    string SearchText,
    EntityId? EntityId = null,
    RelationshipId? RelationshipId = null,
    FieldId? FieldId = null,
    EntityId? SourceEntityId = null,
    EntityId? TargetEntityId = null,
    string? Value = null,
    IReadOnlyList<string>? Aliases = null,
    string? Description = null)
{
    public IReadOnlyList<string> EffectiveAliases => Aliases ?? [];
}